### PR TITLE
Convert view-in-grafana to influxdb

### PR DESCRIPTION
**Pull request by jarretlavallee**
This PR converts the containers used in `view-in-grafana.sh` from graphite to influxdb. The main motivation is to have a consistent platform with puppetlabs/pe_metrics_dashboard for ease of maintenance. The secondary goal is to eliminate the use and management of the `reidmv/grafana-puppetserver` docker image by using the vendors' images. 

This PR modifies the behavior of `view-in-grafana.sh` to use newly defined influxdb and grafana containers and load the data in. The dashboards are direct copies of the archive ones in puppetlabs/pe_metrics_dashboard [here](https://github.com/puppetlabs/puppetlabs-pe_metrics_dashboard/tree/master/files). 

Prior to merging this PR, I want to ensure there is dashboard parity between the graphite dashboards and the new influxdb dashboards. Any suggestions are appreciated.

### CODE REVIEW

**Comment by npwalker** [comment]
@jarretlavallee is it possible to avoid copying the dashboards into here?  Can we create a docker container that applies the module to lay down the dashboards?  

What are the tradeoffs there?

**Comment by jarretlavallee** [comment]
@npwalker Good idea. Maybe we can host them externally and pull them down as part of the script or a submodule? I may even be able to pull them directly from the pe_metrics_dashboard module directly. Ideally we would have a single source for the dashboards.  

My concern with building a container is the management of it. Every time we want to update a dashboard we would build a new container and push it out. Currently we have travis do this and push the container out to docker hub under Reid's namespace based on the version in the dockerfile. I was trying to eliminate this need by loading the dashboards into the official grafana container during runtime. That way we can just update the dashboards without forcing the user pull a new container. It also allows for easier testing of dashboards as they are loaded at runtime.  

If we build a container it would be nice to have it under the puppetlabs namespace. We can probably come up with a methodology to pull the dashboards in and build it automatically with travis as well. It would ensure consistency for anyone using that version of the container, and allow for the dashboards to be directly editable, as with this method they are not.

**Comment by npwalker** [comment]
I think this is just a script that we control we could just do something janky like curl the files down before booting the container.  

The point was just to not have to copy the dashboards around as that'll get old.

**Comment by jarretlavallee** [comment]
@npwalker Sounds good. I just pushed a commit to download the dashboards prior to spinning up the containers.

**Comment by npwalker** [comment]
@jarretlavallee If it's not too difficult I think we should leave a script that spins up graphite and one that spins up influxdb.

**Comment by jarretlavallee** [comment]
@npwalker How about an argument that specifies which to use?

**Comment by npwalker** [comment]
sure, I just figured we should leave both options mostly to allow for testing the conversion script if we have a need.

**Comment by jarretlavallee** [comment]
Updated to add config options for using graphite and influxdb. It defaults to graphite, but influxdb can be enabled with a simple argument. I also threw in an option to build the container locally for ease of development of the graphite dashboards.

**Comment by npwalker** [comment]
@jarretlavallee what do we need to move this forward?

**Comment by jarretlavallee** [comment]
@npwalker The updates complete the work in progress, so It just needs to be reviewed. It would be nice if I could get someone else to test it as well. My testing did not show any issues, but it is always nice to validate.

File: view-in-grafana.sh [hunk 1]
```diff
@@ -1,59 +1,114 @@
 #!/bin/bash
 
+# VARIABLES
+BUILDLOCAL=false
+INFLUXDB=false
```
Comment (npwalker) [inline view-in-grafana.sh]: I thought influxdb was going to be the default and graphite was going to be opt-in?

Comment (jarretlavallee) [reply view-in-grafana.sh]: I added a new commit to default to InfluxDB.

**Comment by jarretlavallee** [comment]
It looks like we do not have dashboard parity between InfluxDB and Graphite, so I'll have to add that as a part of PR.

**Comment by jarretlavallee** [comment]
The dashboards from the puppet_metrics_dashboard module that we are pulling look to be sufficient for the use case of this repo. We can merge this if someone is will to do a quick test to ensure it works the way they would expect.

**Comment by jarretlavallee** [comment]
@gguillotte Do you have time to review this PR with the metrics that you were working with? InfluxDB should be slightly more performant with a large set of data in comparison to graphite.

**Comment by jarretlavallee** [comment]
@tkishel @gguillotte @npwalker Are there any objections on this PR? I plan on merging it next Monday if no one has any concerns. I have been using it for a while now and it seems to work pretty well.

**Comment by tkishel** [comment]
This results in no dashboards for me:

```
git clone --single-branch --branch influxdb_docker  https://github.com/jarretlavallee/puppet-metrics-viewer.git 
cd puppet-metrics-viewer
./view-in-grafana.sh /Users/tom.kishel/Downloads/usaa/puppet_enterprise_support/metrics/puppetserver
```

```
Getting the latest container images
Building local containers
Getting the latest graphs
./view-in-grafana.sh: line 69: wget: command not found
./view-in-grafana.sh: line 70: wget: command not found
./view-in-grafana.sh: line 71: wget: command not found
./view-in-grafana.sh: line 72: wget: command not found
Starting Containers
Creating network "influxdb-grafana_default" with the default driver
Creating influxdb ... done
Creating grafana  ... done
Extracting data from tarballs...
Waiting for database to be ready...
ready
Deleting json files past 30 retention_days...
Deleted        0 files past retention_days
Loading data...

Metrics ready! View at http://localhost:3000
Username: admin
Password: admin

Press enter key to exit...
```

Otherwise, we shoudl spell check the README:

```
peroid -> period
asumption -> assumption

back end -> backend
tar balls -> tar archive
```

**Comment by jarretlavallee** [comment]
thanks @tkishel! It looks like it failed to download the dashboards since `wget` is not on the path. 

~~~
./view-in-grafana.sh: line 69: wget: command not found
./view-in-grafana.sh: line 70: wget: command not found
./view-in-grafana.sh: line 71: wget: command not found
./view-in-grafana.sh: line 72: wget: command not found
~~~

Do you have `curl` on the system? I can change to that as it may be more universal.

**Review by tkishel — APPROVED** [review]
Oh yeah, functions!

**Comment by jarretlavallee** [comment]
@tkishel the minor README edits are up as well. Thanks for reviewing this.

**Pull request merged**